src/java.base/share/classes/java/util/List.java

```diff
@@ -1228,6 +1228,12 @@ static <E> List<E> copyOf(Collection<? extends E> coll) {
      * lazy list for the same index, an {@linkplain IllegalStateException}
      * will be thrown.
      * <p>
+     * The returned list's {@linkplain Object#equals(Object) equals()} and
+     * {@linkplain Object#hashCode() hashCode()} methods may trigger initialization of one
+     * or more lazy elements. The returned list's {@linkplain Object#toString() toString()}
+     * method never triggers initialization of elements. Instead, an
+     * implementation-dependant string is returned for uninitialized elements.
```

Comment:
```suggestion
     * implementation-dependent string is returned for uninitialized elements.
```

Reply:
Design comment: I note some internal discomfort at the fact that equals/hashCode trigger initialization, but toString does not... effectively this would make replacing a regular immutable list with a lazy list not possible in all cases. Not saying we should do anything now -- but something to monitor